cmake/Modules/FindTPLROCM.cmake

```diff
@@ -0,0 +1,11 @@
+include(FindPackageHandleStandardArgs)
+
+FIND_LIBRARY(AMD_HIP_LIBRARY amdhip64 PATHS ENV ROCM_PATH PATH_SUFFIXES lib)
+FIND_LIBRARY(HSA_RUNTIME_LIBRARY hsa-runtime64 PATHS ENV ROCM_PATH PATH_SUFFIXES lib)
+
+find_package_handle_standard_args(TPLROCM DEFAULT_MSG AMD_HIP_LIBRARY HSA_RUNTIME_LIBRARY)
+
+kokkos_create_imported_tpl(ROCM INTERFACE
+  LINK_LIBRARIES ${HSA_RUNTIME_LIBRARY} ${AMD_HIP_LIBRARY}
+  COMPILE_OPTIONS -D__HIP_ROCclr__
```

Comment:
Shouldn't this be a `COMPILE_DEFINITIONS`?

Reply:
Yes you are right for classic cmake but `kokkos_create_imported_tpl` does not currently support it probably because `target_compile_definitions` was only added in 3.11
https://github.com/kokkos/kokkos/blob/233fc53f04ac74e6bdb024fdb6ddf2a947f74eff/cmake/kokkos_functions.cmake#L310-L314